doc/dev/adapter-development-walkthrough.md

```diff
@@ -0,0 +1,807 @@
+* [Step 8: Write test and validate your adapter (optional)](#step-8-write-test-and-validate-your-adapter-optional)
+* [Step 9: Cleanup](#step-9-cleanup)
+* [Step 10: Next](#step-10-next)
+
+
+# Before you start
+
+Download a local copy of the Mixer repo
+
+```
+git clone https://github.com/istio/mixer
+```
+
+Install bazel (version 0.5.2 or higher) from [https://bazel.build/](https://bazel.build/) and add it to your PATH
+
+Set the MIXER_REPO variable to the path where the mixer repository is on the local machine. Example `export MIXER_REPO=$GOPATH/src/istio.io/mixer`
+
+Successfully build the repo.
+
+```
+pushd $MIXER_REPO && bazel build ...
+```
+
+# Step 1: Write basic adapter skeleton code
+
+Create `mysampleadapter` directory and navigate to it.
+
+```
+cd $MIXER_REPO/adapter && mkdir mysampleadapter && cd mysampleadapter
+```
+
+Create the file name mysampleadapter.go with the following content
+
+_It defines the adapter's `builder` and `handler` types
+along with the interfaces required to support the 'metric' template. This code so far does not add any functionality for
+printing details in a file. It is done in later steps._
+
+```
+package mysampleadapter
+
+import (
+  "context"
+
+  "github.com/gogo/protobuf/types"
+  "istio.io/mixer/pkg/adapter"
+  "istio.io/mixer/template/metric"
+)
+
+type (
+  builder struct {
+  }
+  handler struct {
+  }
+)
+
+// ensure types implement the requisite interfaces
+var _ metric.HandlerBuilder = &builder{}
+var _ metric.Handler = &handler{}
+
+///////////////// Configuration-time Methods ///////////////
+
+// adapter.HandlerBuilder#Build
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+  return &handler{}, nil
+}
+
+// adapter.HandlerBuilder#SetAdapterConfig
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+}
+
+// adapter.HandlerBuilder#Validate
+func (b *builder) Validate() (ce *adapter.ConfigErrors) { return nil }
+
+// metric.HandlerBuilder#SetMetricTypes
+func (b *builder) SetMetricTypes(types map[string]*metric.Type) {
+}
+
+////////////////// Request-time Methods //////////////////////////
+// metric.Handler#HandleMetric
+func (h *handler) HandleMetric(ctx context.Context, insts []*metric.Instance) error {
+  return nil
+}
+
+// adapter.Handler#Close
+func (h *handler) Close() error { return nil }
+
+////////////////// Bootstrap //////////////////////////
+// GetInfo returns the adapter.Info specific to this adapter.
+func GetInfo() adapter.Info {
+  return adapter.Info{
+     Name:        "mysampleadapter",
+     Description: "Logs the metric calls into a file",
+     SupportedTemplates: []string{
+        metric.TemplateName,
+     },
+     NewBuilder:    func() adapter.HandlerBuilder { return &builder{} },
+     DefaultConfig: &types.Empty{},
+  }
+}
+```
+
+
+Now, write corresponding BUILD file. Create a file name BUILD with the following content.
+
+<pre>
```

Comment:
In this doc and the other, please replace HTML PRE blocks with backticks.

Also, all bash commands should start with ```bash
All go code should be marked with ```golang

This will let GitHub do proper syntax highlighting.